spec/models/payment_provider_customers/stripe_customer_spec.rb

```diff
@@ -13,4 +13,82 @@
       expect(stripe_customer.payment_method_id).to eq('foo_bar')
     end
   end
+
+  describe '#provider_payment_methods' do
+    subject(:provider_payment_methods) { stripe_customer.provider_payment_methods }
+
+    let(:stripe_customer) { FactoryBot.build_stubbed(:stripe_customer) }
+
+    let(:payment_methods) do
+      described_class::ALLOWED_PAYMENT_METHODS.sample Faker::Number.between(from: 1, to: 2)
+    end
+
+    before { stripe_customer.provider_payment_methods = payment_methods }
+
+    it 'returns provider payment methods' do
+      expect(provider_payment_methods).to eq payment_methods
+    end
+  end
+
+  describe 'validation' do
+    describe 'of provider payment methods' do
+      subject(:valid) { stripe_customer.valid? }
+
+      let(:stripe_customer) do
+        FactoryBot.build_stubbed(:stripe_customer, provider_payment_methods:)
+      end
+
+      let(:errors) { stripe_customer.errors }
+
+      before { subject }
```

Comment:
RSpec/NamedSubject: Name your test subject if you need to reference it explicitly. (https://rspec.rubystyle.guide/#use-subject, https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/NamedSubject)